ScnSoft-Education/LeetCodeTraining.Tests/MergeTwoSortedListsTest.cs

```diff
@@ -0,0 +1,52 @@
+﻿using LeetCodeTraining.MergeTwoSortedLists;
+
+namespace LeetCodeTraining.Tests
+{
+    public class MergeTwoSortedListsSolutionTests
+    {
+        public static ListNode InitializeNode(int[] numbers)
+        {
+            if (numbers.Length == 0)
+                return null;
+
+            ListNode head = new ListNode(numbers[0]);
+            ListNode current = head;
+            for (int i = 1; i < numbers.Length; i++)
+            {
+                current.next = new ListNode(numbers[i]);
+                current = current.next;
+            }
+            return head;
+        }
+
+        public static bool CompareNodes(ListNode node1, ListNode node2)
```

Comment:
Instead this can you try to override Equal method of ListNode class? similar as:
public override bool Equals(object? obj)
In this case you will be able do testing as:

`Assert.Equal(expectedNode, resultNode));`


For this [MemberData(nameof(Data))] can be used to do test method  as:

`public void MergeTwoLists(ListNode list1, ListNode list2, ListNode expected)`

To accomplish using just Assert.Equal(expectedNode, resultNode) it is required to initialize ListNode class similar to:
    ```
public class ListNode
    {
        public int val;
        public ListNode next;

        public ListNode(int val = 0, ListNode next = null)
        {
            this.val = val;
            this.next = next;
        }

        public ListNode(int[] numbers)
        {
            ...
        }

        public override bool Equals(object? obj)
        {
            ...
        }
    }
```